Task6/arrayFunctionCollection.js

```diff
@@ -0,0 +1,116 @@
+const choosingFunc = function()  {
+    //arguments[0] - parametrs (Array*param+)
+    //arguments[1] - function     
+	if( Array.isArray(arguments[0][0]))  {
+        var parameters = Array.prototype.slice.call(arguments[0], 1);
+		const resultArray = arguments[1].call(this, parameters, arguments[0][0]);
+		return resultArray;
+	}
+	const resultFunc = arguments[1].bind(this, ...arguments[0]);
+    this._func[this._func.length] = resultFunc;
+	return this;
+}
+
+var arrayFunctionCollection = {
+
+    _array : [], 
+    _func : [],
+
+	take : function() {
+		return choosingFunc.call(this,
+			arguments,
```

Comment:
I think it's better to pass callback as first arguments ant thencpassing `...arguments` (not just as `arguments`). It allows get rid of ugly `arguments[0][0]` in `choosingFunc`.